desktop/core/src/desktop/js/apps/admin/Overview/Analytics.tsx

```diff
@@ -14,63 +14,106 @@
+import useSaveData from '../../../utils/hooks/useSaveData/useSaveData';
+import LoadingErrorWrapper from '../../../reactComponents/LoadingErrorWrapper/LoadingErrorWrapper';
 import './Overview.scss';
 
-interface UpdatePreferences {
-  status: number;
+interface UsageAnalyticsResponse {
+  analytics_enabled: boolean;
   message?: string;
 }
 
 const Analytics = (): JSX.Element => {
-  const [collectUsage, setCollectUsage] = useState<boolean>(false);
   const { t } = i18nReact.useTranslation();
+  const { data, loading, error, reloadData } = useLoadData<UsageAnalyticsResponse>(
+    GET_USAGE_ANALYTICS_API_URL
+  );
 
-  const saveCollectUsagePreference = async (collectUsage: boolean) => {
-    const response = await post<UpdatePreferences>(ANALYTICS_PREFERENCES_API_URL, {
-      collect_usage: collectUsage
-    });
+  const [analyticsEnabled, setAnalyticsEnabled] = useState<boolean>(false);
 
-    if (response.status === 0) {
-      const successMessage = collectUsage
-        ? t('Analytics have been activated.')
-        : t('Analytics have been deactivated.');
-      huePubSub.publish('hue.global.info', { message: successMessage });
-    } else {
-      const errorMessage = collectUsage
-        ? t('Failed to activate analytics.')
-        : t('Failed to deactivate analytics.');
-      huePubSub.publish('hue.global.error', { message: errorMessage });
+  useEffect(() => {
+    if (data?.analytics_enabled !== undefined) {
+      setAnalyticsEnabled(data.analytics_enabled);
     }
-  };
+  }, [data]);
+
+  const { save: saveAnalyticsPreference, loading: savingPreference } =
+    useSaveData<UsageAnalyticsResponse>(UPDATE_USAGE_ANALYTICS_API_URL, {
+      onSuccess: response => {
+        if (response.analytics_enabled !== undefined) {
+          reloadData();
+          setAnalyticsEnabled(response.analytics_enabled);
+          const successMessage = response.analytics_enabled
+            ? t('Analytics have been activated.')
+            : t('Analytics have been deactivated.');
+          huePubSub.publish('hue.global.info', { message: successMessage });
+        }
+      },
+      onError: error => {
+        console.error('Error updating usage analytics settings:', error);
+        const errorMessage = analyticsEnabled
+          ? t('Failed to deactivate analytics.')
+          : t('Failed to activate analytics.');
+        huePubSub.publish('hue.global.error', { message: errorMessage });
+      },
+      postOptions: {
+        headers: {
+          'Content-Type': 'application/json',
+        }
+      }
+    });
+  
+const handleCheckboxChange = async (event: React.ChangeEvent<HTMLInputElement>) => {
+  const newPreference = event.target.checked;
 
-  const handleCheckboxChange = event => {
-    const newPreference = event.target.checked;
-    setCollectUsage(newPreference);
-    saveCollectUsagePreference(newPreference);
+
+  const payload = {
+    analytics_enabled: newPreference 
   };
 
+  try {
```

Comment:
Don't think you need the try catch here since you catch save the errors in saveAnalyticsPreference already